### PR TITLE
DOC: stats.entropy: expand examples

### PR DESCRIPTION
**Pull request by mdhaber**
Some suggestions to consider. I can remove any you disagree with.  If you're happy with it, please this, and afterwards, I can merge scipy/scipy#16474.

### CODE REVIEW

File: scipy/stats/_entropy.py [hunk 1]
```diff
@@ -19,13 +19,14 @@ def entropy(pk: np.typing.ArrayLike,
             base: Optional[float] = None,
             axis: int = 0
             ) -> Union[np.number, np.ndarray]:
-    """Calculate the entropy of a distribution for given probability values.
+    """
+    Calculate the Shannon entropy/cross entropy of given distribution(s).
```
Comment (mdhaber) [inline scipy/stats/_entropy.py]: To emphasize that this function is for both entropy and relative entropy (improve discoverability).

File: scipy/stats/_entropy.py [hunk 2]
```diff
@@ -19,13 +19,14 @@ def entropy(pk: np.typing.ArrayLike,
...
 
     If only probabilities `pk` are given, the Shannon entropy is calculated as
-    ``S = -sum(pk * log(pk), axis=axis)``.
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:25]: We don't need to use the symbol `S` (which is also the name of the set mentioned in the notes), so let's use the symbol appropriate to the quantity for emphasis. Also, I see that the docstring already showed the `axis` argument. This isn't very informative (and if we're going to show it, why not `base=base`?), so let's simplify.

File: scipy/stats/_entropy.py [hunk 3]
```diff
@@ -46,20 +47,20 @@ def entropy(pk: np.typing.ArrayLike,
 
     Returns
     -------
-    S : {float, array_like}
+    H : {float, array_like}
```
Comment (mdhaber) [inline scipy/stats/_entropy.py]: No need for this to match what's inside the function. However, it would probably be better if it were neither the symbol for entropy or relative entropy; so maybe revert this change and rename the set below?
```suggestion
    S : {float, array_like}
```

File: scipy/stats/_entropy.py [hunk 4]
```diff
@@ -46,20 +47,20 @@ def entropy(pk: np.typing.ArrayLike,
...
         The calculated entropy.
 
     Notes
     -----
     Informally, the Shannon entropy quantifies the expected uncertainty
     inherent in the possible outcomes of a discrete random variable.
-
-    If messages consisting of sequences of symbols from a set S are to be
+    For example,
+    if messages consisting of sequences of symbols from a set ``S`` are to be
```
Comment (mdhaber) [inline scipy/stats/_entropy.py]:
```suggestion
    if messages consisting of sequences of symbols from a set are to be
```
name is not used again, so let's keep `S` as the name of the return above and omit this.

File: scipy/stats/_entropy.py [hunk 5]
```diff
@@ -46,20 +47,20 @@ def entropy(pk: np.typing.ArrayLike,
...
+    H : {float, array_like}
         The calculated entropy.
 
     Notes
     -----
     Informally, the Shannon entropy quantifies the expected uncertainty
     inherent in the possible outcomes of a discrete random variable.
-
-    If messages consisting of sequences of symbols from a set S are to be
+    For example,
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:57]: Two things here: combined first two paragraphs so that Shannon entropy does not appear to be so much more important than relative entropy.

Also, "for example" because this is not the only interpretation.

File: scipy/stats/_entropy.py [hunk 6]
```diff
@@ -46,20 +47,20 @@ def entropy(pk: np.typing.ArrayLike,
...
+    if messages consisting of sequences of symbols from a set ``S`` are to be
     encoded and transmitted over a noiseless channel, then the Shannon entropy
     ``H(pk)`` gives a tight lower bound for the average number of units of
     information needed per symbol if the symbols occur with frequencies
     governed by the discrete distribution `pk` [1]_. The choice of base
-    determines the choice of units, ``e`` for nats, ``2`` for bits, etc.
+    determines the choice of units; e.g. ``e`` for nats, ``2`` for bits, etc.
```
Comment (mdhaber) [inline scipy/stats/_entropy.py]:
```suggestion
    determines the choice of units; e.g., ``e`` for nats, ``2`` for bits, etc.
```

File: scipy/stats/_entropy.py [hunk 7]
```diff
@@ -70,10 +71,11 @@ def entropy(pk: np.typing.ArrayLike,
 
     A related quantity, the cross entropy ``CE(pk, qk)``, satisfies the
     equation ``CE(pk, qk) = H(pk) + D(pk|qk)`` and can also be calculated with
-    the formula ``S = -sum(pk * log(qk), axis=axis)``. It gives the average
+    the formula ``CE = -sum(pk * log(qk))``. It gives the average
     number of units of information needed per symbol if an encoding is
     optimized for the probability distribution `qk` when the true distribution
-    is `pk`.
+    is `pk`. It is not computed directly by `entropy`, but it can be computed
+    using two calls to the function (see Examples).
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:78]: For additional emphasis.

File: scipy/stats/_entropy.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/_entropy.py]:
```suggestion
    Calculate the Shannon entropy/relative entropy of given distribution(s).
```
Oops!

File: scipy/stats/_entropy.py [hunk 8]
```diff
@@ -89,32 +91,40 @@ def entropy(pk: np.typing.ArrayLike,
 
     Examples
     --------
-
-    >>> from scipy.stats import entropy
-
-    Bernoulli trial with different p.
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:95]: Not sure what `p` was originally? So we can remove this.

File: scipy/stats/_entropy.py [hunk 9]
```diff
@@ -89,32 +91,40 @@ def entropy(pk: np.typing.ArrayLike,
...
 
-    >>> entropy([9/10, 1/10], base=2)
+    >>> qk = np.array([9/10, 1/10])  # biased coin
+    >>> entropy(qk, base=base)
     0.46899559358928117
 
-    Relative entropy:
-
-    >>> entropy([1/2, 1/2], qk=[9/10, 1/10])
-    0.5108256237659907
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:109]: Should stick with the same base throughout.

File: scipy/stats/_entropy.py [hunk 10]
```diff
@@ -89,32 +91,40 @@ def entropy(pk: np.typing.ArrayLike,
...
 
-    >>> entropy([1/2, 1/2], base=2)
+    >>> from scipy.stats import entropy
+    >>> base = 2  # work in units of bits
+    >>> pk = np.array([1/2, 1/2])  # fair coin
+    >>> H = entropy(pk, base=base)
+    >>> H
     1.0
+    >>> H == -np.sum(pk * np.log(pk)) / np.log(base)
+    True
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:103]: We could delete these if they are distracting, but I thought they might be helpful.

File: scipy/stats/_entropy.py [hunk 11]
```diff
@@ -89,32 +91,40 @@ def entropy(pk: np.typing.ArrayLike,
...
-
-    >>> from scipy.stats import entropy
-
-    Bernoulli trial with different p.
     The outcome of a fair coin is the most uncertain:
 
-    >>> entropy([1/2, 1/2], base=2)
+    >>> from scipy.stats import entropy
+    >>> base = 2  # work in units of bits
+    >>> pk = np.array([1/2, 1/2])  # fair coin
```
Comment (mdhaber) [inline scipy/stats/_entropy.py:98]: Might as well assign to variables since they're used several times.

**Comment by steppi** [comment]
Looks good! I agree with all of the changes.
